.github/workflows/test.yml

```
@@ -0,0 +1,52 @@
+name: 'Run Tests'
+
+on:
+  push:
+    branches: ['**']
+  pull_request:
+    branches: ['**']
```

Let's run them on PR opened to the main.
```suggestion
on:
  pull_request:
    branches: ['main']
```

---

It's better to keep it for all branches, until we find its cost not worth it (if any)